internal/store/model/device.go

```diff
@@ -53,6 +53,9 @@ func NewDeviceFromApiResource(resource *api.Device) (*Device, error) {
 
 	spec := api.DeviceSpec{}
 	if resource.Spec != nil {
+		if resource.Spec.DecommissionRequested != nil {
+			return nil, flterrors.ErrDecommission
+		}
```

Comment:
Why is this necessary?  I think it would return an error whenever doing anything to any decommissioned device.

Reply:
Yes, that is the intended result as no changes should be made to decommissioned devices.

Reply:
I'm actually removing this check, as it no longer works if we use `Update()`.